pkg/controller/node/nodecontroller.go

```diff
@@ -368,6 +371,28 @@ func (nc *NodeController) maybeDeleteTerminatingPod(obj interface{}) {
 	}
 }
 
+// cleanupOrphanedPods deletes pods that are bound to nodes that don't
+// exist.
+func (nc *NodeController) cleanupOrphanedPods() {
+	pods, err := nc.podStore.List(labels.Everything())
+	if err != nil {
+		utilruntime.HandleError(err)
+		return
+	}
+
+	for _, pod := range pods {
+		if pod.Spec.NodeName == "" {
+			continue
+		}
+		if _, exists, _ := nc.nodeStore.Store.GetByKey(pod.Spec.NodeName); exists {
+			continue
+		}
+		if err := nc.forcefullyDeletePod(pod); err != nil {
```

Comment:
GetByKey, can't return errors stupidly. That should be cleaned up:

https://github.com/kubernetes/kubernetes/blob/master/pkg/client/cache/store.go#L192

I'll do this in a follow up.

These cashes can be stale. There's a race here if a (node get's created -> scheduler get's update of node -> new pod gets assigned to a node -> controller get's update of pod) before (node get's created -> controller get's update of node). It seems less likely and of smaller consequence then the but this fixes. Any suggestion on how to work around this?